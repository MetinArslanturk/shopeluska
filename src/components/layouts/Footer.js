import React from 'react';

const Footer = () => {
    return (
        <>
            <div>
                Â©{' '}
                <a
                    rel="noopener noreferrer"
                    target="_blank"
                    href="https://www.metinarslanturk.com"
                >
                    Metin Arslanturk
                </a>{' '}
                <span className="mail-text">
                    ( contact@metinarslanturk.com )
                </span>{' '}
                --{' '}
                <a
                    rel="noopener noreferrer"
                    target="_blank"
                    href="https://github.com/MetinArslanturk/shopeluska"
                >
                    GitHub of This Project
                </a>
            </div>
        </>
    );
};

export default Footer;
