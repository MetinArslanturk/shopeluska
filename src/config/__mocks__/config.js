const actualConfig = jest.requireActual('../../config/config');

export const history = { push: jest.fn() };
export const baseHref = actualConfig.baseHref;