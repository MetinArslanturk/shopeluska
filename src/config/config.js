import { createBrowserHistory } from 'history';

const history = createBrowserHistory();
const baseHref = '/';

export {history, baseHref};