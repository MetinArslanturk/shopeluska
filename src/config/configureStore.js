import { createStore, combineReducers, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import layoutReducer from '../reducers/layout';
import authReducer from '../reducers/auth';
import productsReducer from '../reducers/products';
import shoppingReducer from '../reducers/shopping';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default () => {
    const store = createStore(
      combineReducers({
        layout: layoutReducer,
        auth: authReducer,
        products: productsReducer,
        shopping: shoppingReducer
      }),
      composeEnhancers(applyMiddleware(thunk))
    );
  
    return store;
  };
  