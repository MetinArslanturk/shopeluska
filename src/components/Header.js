import React, { useState } from 'react';
import { connect } from 'react-redux';
import { history, baseHref } from '../config/config';
import { Drawer, Row, Col, Button, Input } from 'antd';
import { Link } from 'react-router-dom';
import { BarsOutlined } from '@ant-design/icons';
import { startLogout } from '../actions/auth';
import {getUserName, isAuthenticated} from '../selectors/auth';
const { Search } = Input;

const clickLogin = () => {
    history.push(baseHref + 'login');
}

const clickMyAccount = () => {
    history.push(baseHref + 'my-account/my-profile');
}

export const Header = ({ isAuthenticated, username, startLogout}) => {
    const [visibleNav, setVisibleNav] = useState(false);

    return (
        <>
            <div className="header-wrapper">

                <Row>
                    <Col span={4}>
                        <div className="brand-logo">
                            <Link to={baseHref}><img alt="logo" className="logo-image" src="logo.png" /></Link>
                        </div>
                    </Col>
                    <Col xs={4} sm={4} md={7} lg={8} xl={12}>
                        <div className="search-box">
                            <Search placeholder="input search text" onSearch={value => console.log(value)} enterButton />
                        </div>
                    </Col>
                    <Col xs={4} sm={6} md={6} lg={6} xl={4}>
                        <div className="user-nav">
                            {!isAuthenticated && (
                                <Button type="primary" className="login-button" onClick={clickLogin}>Login</Button>
                            )}
                            {isAuthenticated && (
                                <>
                                    <Button className="my-account-button" type="primary" onClick={clickMyAccount}>My Account</Button>
                                    <Button className="logout-button" onClick={startLogout}>Logout</Button>
                                </>
                            )}
                        </div>
                        <div className="head-menu-toggle-button">
                            <BarsOutlined onClick={e => setVisibleNav(true)} style={{ fontSize: '32px', marginTop: '18px' }} />
                        </div>
                    </Col>


                </Row>
            </div>
            <Drawer
                title="Shopeluska Navigation"
                placement="right"
                closable={true}
                onClose={() => {
                    setVisibleNav(false);
                }}
                visible={visibleNav}
            >
                <p>Some contents...</p>
                <p>Some contents...</p>
                <p>Some contents...</p>
            </Drawer>
        </>
    )
}
const mapStateToProps = (state) => ({
    isAuthenticated: isAuthenticated(state.auth.user),
    username: getUserName(state.auth.user)
});

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(mapStateToProps, mapDispatchToProps)(Header);