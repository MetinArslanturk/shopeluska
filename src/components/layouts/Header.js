import React, { useState } from 'react';
import { connect } from 'react-redux';
import { history, baseHref } from '../../config/config';
import { Drawer, Row, Col, Button, Input, Badge } from 'antd';
import { Link } from 'react-router-dom';
import { BarsOutlined, ShoppingCartOutlined } from '@ant-design/icons';
import { startLogout } from '../../actions/auth';
import {isAuthenticated} from '../../selectors/auth';
const { Search } = Input;

const clickLogin = () => {
    history.push(baseHref + 'login');
}

const clickMyAccount = () => {
    history.push(baseHref + 'my-account');
}

export const Header = ({ isAuthenticated, startLogout}) => {
    const [visibleNav, setVisibleNav] = useState(false);

    return (
        <>
            <div className="header-wrapper">

                <Row>
                    <Col span={4}>
                        <div className="brand-logo">
                            <Link to={baseHref}><img alt="logo" className="logo-image" src={baseHref + 'logo.png'} /></Link>
                        </div>
                    </Col>
                    <Col xs={4} sm={4} md={7} lg={8} xl={12}>
                        <div className="search-box">
                            <Search placeholder="input search text" onSearch={value => console.log(value)} enterButton />
                        </div>
                    </Col>
                    <Col xs={4} sm={6} md={6} lg={6} xl={4}>
                        <div className="user-nav">
                        <Badge className="badge-text" count={10}>
                        <ShoppingCartOutlined style={{ fontSize: '36px'}}/>
                        </Badge>
                        
                            {!isAuthenticated && (
                                <Button type="primary" className="login-button" onClick={clickLogin}>Login</Button>
                            )}
                            {isAuthenticated && (
                                <>
                                    <Button className="my-account-button my-account" type="primary" onClick={clickMyAccount}>My Account</Button>
                                    <Button className="logout-button" onClick={startLogout}>Logout</Button>
                                </>
                            )}
                        </div>
                        <div className="head-menu-toggle-button">
                            <BarsOutlined onClick={e => setVisibleNav(true)} style={{ fontSize: '32px', marginTop: '18px' }} />
                        </div>
                    </Col>


                </Row>
            </div>
            <Drawer
                title="Shopeluska Navigation"
                placement="right"
                closable={true}
                onClose={() => {
                    setVisibleNav(false);
                }}
                visible={visibleNav}
            >
                <p>Some contents...</p>
                <p>Some contents...</p>
                <p>Some contents...</p>
            </Drawer>
        </>
    )
}
const mapStateToProps = (state) => ({
    isAuthenticated: isAuthenticated(state.auth.user)
});

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(mapStateToProps, mapDispatchToProps)(Header);