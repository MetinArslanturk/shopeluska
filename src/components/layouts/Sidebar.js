import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Layout } from 'antd';
import { NavLink } from 'react-router-dom';
import { baseHref } from '../../config/config';
import { menuItems  } from "../../helpers/menuItems";

const { Sider } = Layout;

const Sidebar = ({ showSidebar }) => {

    return (
        <>
            {showSidebar && (
                <Sider breakpoint="lg" width={230} collapsedWidth="0">
                    <nav className="nav">
                        <ul>
                            {menuItems.map((item) => {
                                return (
                                    <li key={item.content}>
                                        <NavLink
                                            activeClassName="active"
                                            exact
                                            to={baseHref + item.link}
                                        >
                                            {item.content}
                                        </NavLink>
                                    </li>
                                );
                            })}
                        </ul>
                    </nav>
                </Sider>
            )}
        </>
    );
};

Sidebar.propTypes = {
    showSidebar: PropTypes.bool,
};

const mapStateToprops = (state) => ({
    showSidebar: state.layout.showSidebar,
});
export default connect(mapStateToprops)(Sidebar);
