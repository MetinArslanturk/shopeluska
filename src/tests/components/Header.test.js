import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/layouts/Header';
import { history, baseHref } from '../../config/config';

jest.mock('../../config/config');

test('should render header when non-authenticated', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(
        <Header isAuthenticated={false} startLogout={startLogout} />
    );
    expect(wrapper).toMatchSnapshot();
});

test('should trigger history push when non-authenticated and click login', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(
        <Header isAuth={false} startLogout={startLogout} />
    );
    wrapper.find('.login-button').prop('onClick')();
    expect(history.push).toHaveBeenLastCalledWith(`${baseHref}login`);
});

test('should render header when authenticated', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header isAuth logout={startLogout} />);
    expect(wrapper).toMatchSnapshot();
});

test('should trigger startlogout when authenticated and click logout', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header isAuth logout={startLogout} />);
    wrapper.find('.logout-button').prop('onClick')();

    expect(startLogout).toHaveBeenCalled();
});

test('should trigger history push when authenticated and click my account', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header isAuth logout={startLogout} />);
    wrapper.find('.my-account-button').prop('onClick')();

    expect(history.push).toHaveBeenLastCalledWith(`${baseHref}my-account`);
});

test('should trigger pushing history when search', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header isAuth logout={startLogout} />);
    wrapper.find('.search-c').prop('onSearch')('test');

    expect(history.push).toHaveBeenLastCalledWith(`${baseHref}search?key=test`);
});
