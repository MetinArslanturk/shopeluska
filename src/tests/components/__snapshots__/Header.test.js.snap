// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header when authenticated 1`] = `
<Fragment>
  <div
    className="header-wrapper"
  >
    <Row
      gutter={0}
    >
      <Col
        span={4}
      >
        <div
          className="brand-logo"
        >
          <Link
            to="/shopeluska/"
          >
            <img
              alt="logo"
              className="logo-image"
              src="/shopeluska/logo.png"
            />
          </Link>
        </div>
      </Col>
      <Col
        lg={8}
        md={7}
        sm={4}
        xl={12}
        xs={4}
      >
        <div
          className="search-box"
        >
          <Search
            className="search-c"
            enterButton={true}
            onSearch={[Function]}
            placeholder="input search text"
          />
        </div>
      </Col>
      <Col
        lg={6}
        md={6}
        sm={6}
        xl={4}
        xs={4}
      >
        <div
          className="user-nav"
        >
          <Link
            to="/shopeluska/shopping-cart"
          >
            <Connect(ShoppingCartIcon) />
          </Link>
          <Button
            block={false}
            className="my-account-button my-account"
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={[Function]}
            type="primary"
          >
            My Account
          </Button>
          <Button
            block={false}
            className="logout-button"
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={[MockFunction]}
          >
            Logout
          </Button>
        </div>
        <div
          className="head-menu-toggle-button"
        >
          <ForwardRef(BarsOutlined)
            onClick={[Function]}
            style={
              Object {
                "fontSize": "32px",
                "marginTop": "18px",
              }
            }
          />
        </div>
      </Col>
    </Row>
  </div>
  <withConfigConsumer(Drawer)
    closable={true}
    onClose={[Function]}
    placement="right"
    title="Shopeluska Navigation"
    visible={false}
  >
    <ul>
      <li
        key="Homepage"
      >
        <NavLink
          activeClassName="active"
          exact={true}
          to="/shopeluska/"
        >
          Homepage
        </NavLink>
      </li>
      <li
        key="Clothing"
      >
        <NavLink
          activeClassName="active"
          exact={true}
          to="/shopeluska/category/clothing"
        >
          Clothing
        </NavLink>
      </li>
      <li
        key="Shoes"
      >
        <NavLink
          activeClassName="active"
          exact={true}
          to="/shopeluska/category/shoes"
        >
          Shoes
        </NavLink>
      </li>
      <li
        key="Accessories"
      >
        <NavLink
          activeClassName="active"
          exact={true}
          to="/shopeluska/category/accessories"
        >
          Accessories
        </NavLink>
      </li>
    </ul>
  </withConfigConsumer(Drawer)>
</Fragment>
`;

exports[`should render header when non-authenticated 1`] = `
<Fragment>
  <div
    className="header-wrapper"
  >
    <Row
      gutter={0}
    >
      <Col
        span={4}
      >
        <div
          className="brand-logo"
        >
          <Link
            to="/shopeluska/"
          >
            <img
              alt="logo"
              className="logo-image"
              src="/shopeluska/logo.png"
            />
          </Link>
        </div>
      </Col>
      <Col
        lg={8}
        md={7}
        sm={4}
        xl={12}
        xs={4}
      >
        <div
          className="search-box"
        >
          <Search
            className="search-c"
            enterButton={true}
            onSearch={[Function]}
            placeholder="input search text"
          />
        </div>
      </Col>
      <Col
        lg={6}
        md={6}
        sm={6}
        xl={4}
        xs={4}
      >
        <div
          className="user-nav"
        >
          <Link
            to="/shopeluska/shopping-cart"
          >
            <Connect(ShoppingCartIcon) />
          </Link>
          <Button
            block={false}
            className="login-button"
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={[Function]}
            type="primary"
          >
            Login
          </Button>
        </div>
        <div
          className="head-menu-toggle-button"
        >
          <ForwardRef(BarsOutlined)
            onClick={[Function]}
            style={
              Object {
                "fontSize": "32px",
                "marginTop": "18px",
              }
            }
          />
        </div>
      </Col>
    </Row>
  </div>
  <withConfigConsumer(Drawer)
    closable={true}
    onClose={[Function]}
    placement="right"
    title="Shopeluska Navigation"
    visible={false}
  >
    <ul>
      <li
        key="Homepage"
      >
        <NavLink
          activeClassName="active"
          exact={true}
          to="/shopeluska/"
        >
          Homepage
        </NavLink>
      </li>
      <li
        key="Clothing"
      >
        <NavLink
          activeClassName="active"
          exact={true}
          to="/shopeluska/category/clothing"
        >
          Clothing
        </NavLink>
      </li>
      <li
        key="Shoes"
      >
        <NavLink
          activeClassName="active"
          exact={true}
          to="/shopeluska/category/shoes"
        >
          Shoes
        </NavLink>
      </li>
      <li
        key="Accessories"
      >
        <NavLink
          activeClassName="active"
          exact={true}
          to="/shopeluska/category/accessories"
        >
          Accessories
        </NavLink>
      </li>
    </ul>
  </withConfigConsumer(Drawer)>
</Fragment>
`;
